.github/workflows/run_tests.yml

```diff
@@ -19,6 +19,11 @@ env:
   # (TravisCI quietly defined this on all their platforms, but we have to give it manually on GithubCI.)
   DEBIAN_FRONTEND: 'noninteractive'
   HDF5_USE_FILE_LOCKING: 'FALSE'
+  # Skip to the headless matplotlib renderer, which is less
+  # bug-prone in the constrained environment of CI
+  # Tip from a matplotlib dev: https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3388#issuecomment-846091012
+  # Ref: https://matplotlib.org/stable/users/explain/backends.html
+  MPLBACKEND: 'Agg'
```

Comment:
Could this be relevant, possibly? 

https://github.com/ivadomed/ivadomed/blob/bd904e5e139fa6a437abfc225216f6057824f1e3/ivadomed/utils.py#L225

Reply:
Thank you for your eyes! When I got to that point I was just tired of the problem, so I'm really happy you picked up my baton to keep us going.

It could be either. `use()` _sets a **global**_, so its leaking into other tests, [e.g.](https://github.com/ivadomed/ivadomed/runs/6480501528?check_suite_focus=true):

```
ivadomed/metrics.py:305: in plot_roc_curve
    plt.figure()
[...]
../../../hostedtoolcache/Python/3.8.12/x64/lib/python3.8/tkinter/__init__.py:2272: in __init__
    self._loadtk()
```

```
testing/unit_tests/test_mixup.py:27: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
ivadomed/mixup.py:36: in mixup
    save_mixup_sample(ofolder, data, targets, lambda_tensor)
ivadomed/mixup.py:63: in save_mixup_sample
    plt.figure(figsize=(20, 10))
[...]
../../../hostedtoolcache/Python/3.8.12/x64/lib/python3.8/tkinter/__init__.py:2272: in __init__
    self._loadtk()
```

```
 testing/functional_tests/test_visualize_transforms.py:21: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
ivadomed/scripts/visualize_transforms.py:183: in main
    run_visualization(input=args.input, config=args.config, number=int(args.number),
ivadomed/scripts/visualize_transforms.py:171: in run_visualization
    imed_utils.plot_transformed_sample(before,
ivadomed/utils.py:210: in plot_transformed_sample
    plt.figure(figsize=(20, 10))
[...]
../../../hostedtoolcache/Python/3.8.12/x64/lib/python3.8/tkinter/__init__.py:2272: in __init__
    self._loadtk()
```

None of those three `figure()` calls are near the `use()` call but it's still up in their business.

Reply:
Nice! Those were indeed the final pieces. CI is green: https://github.com/ivadomed/ivadomed/actions/runs/2344760419

![Screenshot 2022-05-18 at 06-47-36 Use headless ( Agg ) matplotlib backend · ivadomed_ivadomed@23e018f](https://user-images.githubusercontent.com/987487/169021844-da46e3eb-ab1c-4c65-92f9-a04ceaead117.png)